### ISSUE TITLE
Composition with circular reference causes infinite loop

### ISSUE THREAD

**Issue by philschleier** [daveshanley/vacuum#459]
The following spec causes vacuum to spin infinitely and consume all available memory:

```
openapi: "3.0.1"

info:
  title: Test
  version: 1.0.0

paths: {}

components:
  schemas:
    A:
      type: object
      properties: {}

    B:
      type: object
      allOf:
        - $ref: '#/components/schemas/A'
      properties:
        children:
          type: array
          items:
            $ref: '#/components/schemas/B'
```

Not sure what to call it, not even sure if it's valid, but probably shouldn't break vacuum...

**Comment by daveshanley** [comment]
which version of vacuum did you test against?

**Comment by philschleier** [comment]
v0.9.7

![image](https://github.com/daveshanley/vacuum/assets/9532061/d7bfaf03-d319-47a0-87a7-e2549b9bd604)

**Comment by daveshanley** [comment]
ok, this is nuts. I will fix.

**Comment by daveshanley** [comment]
This problem was in `libopenapi`. In `v0.15.12` it was fixed, the resolver was not continuing down a path correctly once finding a sibling node that uses a reference. 

I added a test to make sure this loop is closed permanently. 

https://github.com/pb33f/libopenapi/blob/main/index/resolver_test.go#L1339

The upstream fix is available in `v0.9.8` of vacuum.
